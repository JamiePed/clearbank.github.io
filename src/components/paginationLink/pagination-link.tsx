import React from 'react'

import * as Types from './pagination-link.types'
import * as Styles from './pagination-link.styles'

const content = {
  next: {
    label: 'Next',
    icon: 'cb-icon-arrow-right'
  },
  previous: {
    label: 'Previous',
    icon: 'cb-icon-arrow-left'
  }
}

const PaginationLink: React.FunctionComponent<Types.PaginationLinkProps> = ({
  slug,
  title,
  direction
}): JSX.Element => {
  if (!(direction in content)) {
    return null
  }

  const { label, icon } = content[direction]

  return (
    <Styles.Container direction={direction}>
      <Styles.Label>{label}</Styles.Label>
      <Styles.Link to={slug} direction={direction}>
        <Styles.IconWrapper>
          <Styles.Icon className={icon} />
        </Styles.IconWrapper>
        <Styles.Text>{title}</Styles.Text>
      </Styles.Link>
    </Styles.Container>
  )
}

export default PaginationLink
