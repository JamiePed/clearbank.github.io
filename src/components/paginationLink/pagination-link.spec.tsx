import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'

import Component from './pagination-link'

import * as Types from './pagination-link.types'

let component: any = null
const defaultProps:Types.PaginationLinkProps = {
  slug: '/slug/',
  title: 'Slug',
  direction: 'next'
}

beforeEach(() => {
  component = render(<Component {...defaultProps} />)
})

describe('Component', () => {
  test('matches snapshot', () => {
    expect(component.asFragment()).toMatchSnapshot()
  })
})
