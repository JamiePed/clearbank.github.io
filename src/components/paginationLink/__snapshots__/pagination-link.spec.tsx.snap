// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<DocumentFragment>
  .c1 {
  font-size: 14px;
  cursor: default;
  color: #6c7f98;
}

.c5 {
  margin-left: 10px;
  margin-right: 10px;
  font-size: 16px;
}

.c4 {
  width: 22px;
  height: 22px;
  font-size: 25px;
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  will-change: transform;
  -webkit-transition: -webkit-transform 0.25s ease-out;
  -webkit-transition: transform 0.25s ease-out;
  transition: transform 0.25s ease-out;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  color: #3048D1;
  font-weight: bold;
  font-family: "Hurme Geometric Sans 1",sans-serif;
  -webkit-flex-direction: row-reverse;
  -ms-flex-direction: row-reverse;
  flex-direction: row-reverse;
}

.c2:hover {
  color: #0D326E;
}

.c0 {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-align: right;
}

.c0:hover .c3 {
  -webkit-transform: translateX(5px);
  -ms-transform: translateX(5px);
  transform: translateX(5px);
}

<div
    class="c0"
    direction="next"
  >
    <span
      class="c1"
    >
      Next
    </span>
    <a
      class="c2"
      direction="next"
      href="/slug/"
    >
      <span
        class="c3 c4"
      >
        <i
          class="cb-icon-arrow-right"
        />
      </span>
      <span
        class="c5"
      >
        Slug
      </span>
    </a>
  </div>
</DocumentFragment>
`;
